demo/src/screens/componentScreens/SharedTransitionScreen.js

```diff
@@ -0,0 +1,178 @@
+        this.setState({itemId: id, itemLayout});
+
+        setTimeout(() => {
+          this.setState({showDetails: true});
+        }, ANIMATION_DURATION);
+      });
+    }
+  };
+
+  backToList = () => {
+    this.setState({showDetails: false});
+
+    setTimeout(() => {
+      this.setState({itemId: undefined, itemLayout: undefined});
+    }, ANIMATION_DURATION);
+  };
+
+  renderItem = ({item}) => {
+    return (
+      <View marginB-20 padding-20 row height={280}>
+        <Card padding-20 marginV-20 width={'60%'} style={{zIndex: 10}} spread>
+          <View>
+            <Text text70 dark10>
+              {item.title}
+            </Text>
+            <Text text80 dark30>
+              {item.location}
+            </Text>
+          </View>
+          <View>
+            <Text text70 dark30>
+              <Text text50 purple30>
+                {item.views}
+              </Text>
+              {' Views'}
+            </Text>
+          </View>
+        </Card>
+        <SharedTransition.Source
+          id={item.id}
+          data={item}
+          style={[{overflow: 'hidden', position: 'absolute', right: 20, top: 20, bottom: 20, width: 150, height: 200}]}
+        >
+          {this.renderSharedElement(item)}
+        </SharedTransition.Source>
+      </View>
+    );
+  };
+
+  renderList() {
+    return <FlatList data={DATA} renderItem={this.renderItem} keyExtractor={item => item.title} />;
+  }
+
+  renderSharedElement(item = {}) {
+    return (
+      <View flex>
+        <Image
+          style={{width: '100%', height: '100%', borderRadius: 10}}
+          source={{
+            uri: item.image,
+          }}
+        />
+      </View>
+    );
+  }
+
+  renderDetails = data => {
+    if (data) {
+      return (
+        <View flex bg-blue70 pointerEvents="box-none" padding-20 paddingT-100>
+          <View row>
+            <SharedTransition.Target style={{width: 200, height: 250}}>
+              {this.renderSharedElement(data, true)}
+            </SharedTransition.Target>
+            <View flex paddingL-20>
+              <Text text50>{data.title}</Text>
+              <Text text90 purple30 marginT-10>
+                {data.views} Views
+              </Text>
+            </View>
+          </View>
+          <View left marginT-20>
+            <Text text60>{data.location}</Text>
+            <Text>
+              when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has
+              survived not only five centuries, but also the leap into electronic typesetting, remaining essentially
+              unchanged
+            </Text>
+            <Button link marginT-20 dark10 label="Learn More" size="small" />
+          </View>
+        </View>
+      );
+    }
+  };
+
+  render() {
+    return (
+      <View flex dark80>
+        <SharedTransition.Area renderDetails={this.renderDetails}>{this.renderList()}</SharedTransition.Area>
```

Comment:
SharedTarnsition.Area must wrap the list or could it wrap just the list item?

Reply:
it wraps the whole screen, cause it contains the shared context between the shared elements